server/start.go

```diff
@@ -353,34 +359,21 @@ func startInProcess(svrCtx *Context, clientCtx client.Context, appCreator types.
 		return g.Wait()
 	}
 
-	// In case the operator has both gRPC and API servers disabled, there is
-	// nothing blocking this root process, so we need to block manually, so we'll
-	// create an empty blocking loop.
-	g.Go(func() error {
-		<-ctx.Done()
-		return nil
-	})
-
 	// deferred cleanup function
 	// TODO: Make a generic cleanup function that takes in several func(), and runs them all.
 	// then we defer that.
-	defer func() {
-		if tmNode != nil && tmNode.IsRunning() {
-			_ = tmNode.Stop()
-			_ = app.Close()
-		}
-
-		if traceWriterCleanup != nil {
-			traceWriterCleanup()
-		}
-	}()
```

Comment:
the big defer function is broken up into smaller pieces, and put right after the respective resource is created.